### PR TITLE
Merge ROS1 Dockerfiles into a single one

### LINKED ISSUE #40: Missing a <depend> / <build_depend> on asio
```
In file included from /usr/include/websocketpp/transport/asio/base.hpp:31,
                 from /usr/include/websocketpp/transport/asio/connection.hpp:31,
                 from /usr/include/websocketpp/transport/asio/endpoint.hpp:32,
                 from /usr/include/websocketpp/config/asio.hpp:32,
                 from /ros_ws/src/ros-foxglove-bridge/foxglove_bridge_base/src/foxglove_bridge.cpp:4:
/usr/include/websocketpp/common/asio.hpp:46:14: fatal error: asio/version.hpp: No such file or directory
   46 |     #include <asio/version.hpp>
      |              ^~~~~~~~~~~~~~~~~~
```

### PR DESCRIPTION
**Pull request by achim-k**
**Public-Facing Changes**
None

**Description**
Merges the two existing ROS1 Dockerfiles into a single Dockerfile that can be used to build both for `melodic` and `noetic`.

Fixes #40

### CODE REVIEW

File: Dockerfile.ros1 [hunk 1]
```diff
@@ -1,44 +1,31 @@
...
+COPY foxglove_bridge_base /ros1_ws/src/ros-foxglove-bridge/foxglove_bridge_base
+COPY ros1_foxglove_bridge /ros1_ws/src/ros-foxglove-bridge/ros1_foxglove_bridge
 
-# Create a Catkin workspace
-RUN mkdir -p /ros1_ws/src
+# Build and install gtest.
+RUN cd /usr/src/gtest \
+  && cmake CMakeLists.txt \
+  && make \
+  && cp ./*.a /usr/lib || true
```
Comment (achim-k) [inline Dockerfile.ros1:25]: This is required for melodic, I'm not sure why we need this, I believe that it should also work without it.

File: Dockerfile.ros1 [hunk 2]
```diff
@@ -1,44 +1,31 @@
 ARG ROS_DISTRIBUTION=noetic
-FROM ros:$ROS_DISTRIBUTION-ros-core
+FROM ros:$ROS_DISTRIBUTION-ros-base
```
Comment (jhurliman) [inline Dockerfile.ros1:2]: What extra packages are needed from base?

Comment (achim-k) [reply Dockerfile.ros1:2]: Primarily `rosdep`. If `rosdep` is already installed, we do not have to manually decide if we have to install `python-rosdep` (melodic) or `python3-rosdep` (noetic)

File: package.xml [hunk 3]
```diff
@@ -28,6 +28,7 @@
     <test_depend condition="$ROS_VERSION == 2">ament_cmake_gtest</test_depend>
 
     <!-- Common dependencies -->
+    <build_depend>asio</build_depend>
```
[the remark this answers is not in the document]
Comment (achim-k) [reply package.xml:31]: `asio` is the correct rosdep identifier:

```
$ rosdep resolve libasio-dev
ERROR: no rosdep rule for 'libasio-dev'
$ rosdep resolve asio
#apt
libasio-dev
```

And `libasio-dev` is a header only library, hence only the `build_depend`

**Pull request merged**